### ISSUE TITLE
Exception when constructing a 1x1 `:vectorz` array.

### ISSUE THREAD

**Issue by harold** [mikera/core.matrix#302]
With `[net.mikera/core.matrix "0.54.0"]` or `[net.mikera/core.matrix "0.56.0"]`

```
> (m/array :vectorz [(double-array [17])])
ClassCastException [D cannot be cast to java.lang.Number  clojure.lang.RT.doubleCast (RT.java:1312)
```

@mikera --- Should this work? I assume I'm doing something silly.

**Comment by mikera** [comment]
Confirmed this is a subtle bug in core.matrix - will release a fix shortly

**Comment by mikera** [comment]
Fixed in 0.57.0
